test/integration/thruster.cc

```diff
@@ -331,13 +331,12 @@ void ThrusterTest::TestWorld(const std::string &_world,
     EXPECT_NEAR(0.0, angVel.Y(), _baseTol);
     EXPECT_NEAR(0.0, angVel.Z(), _baseTol);
   }
-
+  auto latest_pose = modelPoses.back();
```

Comment:
Even before this point, are we validating that modelPoses has something in it?  I agree this definitely fixes the bug below, just want to make sure that we aren't adding UB here.

Reply:
Perhaps just an `ASSERT_FALSE(modelPoses.empty())` would be sufficient.

Reply:
I guess that's what's desired.
I have put a if statement so that it only clears when the model poses Is empty.
if I have misunderstood anything do let me know.

Reply:
A possible solution for this can be -
modelPoses.clear();
```
modelPoses.clear()
propellerAngVels.clear();
propellerLinVels.clear();
if(!modelPoses.empty())
 {
    auto latest_pose = modelPoses.back();
 }
```
But then the if statement would always be false.
Hence it wouldn't work for that reason. I have put the latest_pose before modelPoses.clear()

Reply:
or else we can do this - 
ASSERT_FALSE(modelPoses.empty())
<<"No modelPoses found in library [" <<  pathToLib << "]";
auto latest_pose = modelPoses.back();
modelPoses.clear()
propellerAngVels.clear();
propellerLinVels.clear();

Reply:
hii, @mjcarroll  any comments ?